scripts/sysusers.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: true; tab-width: 4; -*-
+
+# Creates system users and groups, based on files in the format described
+# in sysusers.d(5). Replacement for systemd-sysusers(8) for the use of RPM
+#
+# USAGE:
+#
+#   sysusers.sh [--root ROOTDIR] [--replace CONFIGFILE] [CONFIGFILE...]
+#
+# SEE ALSO systemd-sysusers(8)
+#
+# Based on sysusers.generate-pre.sh from the Fedora systemd package
+
+SYSTEMDIR=/usr/lib/sysusers.d
+RUNTIMEDIR=/run/sysusers.d
+SYSTEMCONFDIR=/etc/sysusers.d
+
+ROOT=/
+REPLACE=#
+
+while test $# -gt 0 ; do
+	case "${1}" in
+		--root )
+			shift
+			ROOT=${1}
+			shift
+			;;
+		--replace )
+			shift
+			REPLACE=${1}
+			shift
+			;;
+		--help )
+			cat <<EOF
+
+sysusers.sh - Creates system users and groups, based on files in the
+format described in sysusers.d(5). Replacement for systemd-sysusers(8)
+for the use of RPM.
+
+USAGE:
+
+	sysusers.sh [--root ROOTDIR] [--replace CONFIGFILE] [CONFIGFILE...]
+
+SEE ALSO systemd-sysusers(8)
+
+EOF
+			shift
+			;;
+		* )
+			break
+			;;
+	esac
+done
+
+hasuser() {
+	grep "^${1}:" "$ROOT"/etc/passwd >/dev/null
+}
+
+hasuid() {
+	cut -d: -f3 "$ROOT"/etc/passwd | grep "${1}"  >/dev/null
+}
+
+hasgroup() {
+	grep "^${1}:" "$ROOT"/etc/group >/dev/null
+}
+
+user() {
+	local user="$1"
+	local uid="$2"
+	local desc="$3"
+	local group="$4"
+	local home="$5"
+	local shell="$6"
+
+	[ "$desc" = '-' ] && desc=
+	{ [ "$home" = '-' ] || [ "$home" = '' ]; } && home=/
+	{ [ "$shell" = '-' ] || [ "$shell" = '' ]; } && shell=/usr/sbin/nologin
```

Comment:
This needs to be `/sbin/login`. We cannot assume it's in `/usr`.

Reply:
Fix in 077afb85925d04067ebd38bfa520cec0e36f5bbf